cf-deploy-teardown/tasks/k8s-ready-state-check.sh

```diff
@@ -1,72 +1,126 @@
 }
 
 function red() {
-  awk '{ print "\033[31mConfiguration problem detected: " $0 "\033[0m" }';
+    printf "\033[31m%b\033[0m\n" "$1"
+}
+
+function verified() {
+    green "Verified: $1"
+}
+
+function trouble() {
+    red "Configuration problem detected: $1"
 }
 
 function status() {
-  if [ $? -eq 0 ]; then
-    echo "$1" | green
-  else
-    echo "$1" | red
-    FAILED=1
-  fi
+    if [ $? -eq 0 ]; then
+	verified "$1"
+    else
+	trouble "$1"
+	FAILED=1
+    fi
 }
+
+function having_category() {
+    case "$@" in
+	*${category}*)
+	    return 0
+	    ;;
+    esac
+    return 1
+}
+
+echo "Testing $(green $category)"
+
 # cgroup memory & swap accounting in /proc/cmdline
-grep -wq "cgroup_enable=memory" /proc/cmdline
-status "cgroup_enable memory"
+if having_category all node ; then
+    grep -wq "cgroup_enable=memory" /proc/cmdline
+    status "cgroup_enable memory"
 
-grep -wq "swapaccount=1" /proc/cmdline
-status "swapaccount enable"
+    grep -wq "swapaccount=1" /proc/cmdline
+    status "swapaccount enable"
+fi
 
 # docker info should show overlay2
-
-docker info 2> /dev/null | grep -wq "Storage Driver: overlay2"
-status "docker info should show overlay2"
+if having_category all api kube node ; then
+    docker info 2> /dev/null | grep -wq "Storage Driver: overlay2"
+    status "docker info should show overlay2"
+fi
 
 # kube-dns shows 4/4 ready
-
-kubectl get pods --namespace=kube-system --selector k8s-app=kube-dns | grep -Eq '([0-9])/\1 *Running'
-status "kube-dns should shows 4/4 ready"
+if having_category all kube ; then
+    kubectl get pods --namespace=kube-system --selector k8s-app=kube-dns | grep -Eq '([0-9])/\1 *Running'
+    status "kube-dns should shows 4/4 ready"
+fi
 
 # ntp is installed and running
-
-systemctl is-active ntpd >& /dev/null || systemctl is-active systemd-timesyncd >& /dev/null
-status "ntp or systemd-timesyncd must be installed and active"
+if having_category all api kube node ; then
+    systemctl is-active ntpd >& /dev/null || systemctl is-active systemd-timesyncd >& /dev/null
+    status "ntp or systemd-timesyncd must be installed and active"
+fi
 
 # "persistent" storage class exists in K8s
-
-kubectl get storageclasses |& grep -wq "persistent"
-status "'persistent' storage class should exist in K8s"
+if having_category all kube ; then
+    test $(kubectl get storageclasses |& wc -l) -gt 1
+    status "A storage class should exist in K8s"
+fi
 
 # privileged pods are enabled in K8s
-
-kube_apiserver=$(systemctl status kube-apiserver -l | grep "/usr/bin/hyperkube apiserver" )
-[[ $kube_apiserver == *"--allow-privileged"* ]]
-status "Privileged must be enabled in 'kube-apiserver'"
-
-kubelet=$(systemctl status kubelet -l | grep "/usr/bin/hyperkube kubelet" )
-[[ $kubelet == *"--allow-privileged"* ]]
-status "Privileged must be enabled in 'kubelet'"
+if having_category all api ; then
+    kube_apiserver=$(systemctl status kube-apiserver -l | grep "/usr/bin/hyperkube apiserver" )
```

Comment:
Arbitrary kubernetes clusters might not be running the API server as the systemd service `kube-apiserver`.  And they might not put hyperkube at `/usr/bin/`.